src/least_satisfying.rs

```diff
@@ -8,7 +8,7 @@
 use std::collections::BTreeMap;
 use std::fmt;
 
-pub fn least_satisfying<T, P>(slice: &[T], mut predicate: P) -> usize
+pub fn least_satisfying<T, P>(slice: &[T], start_check: bool, mut predicate: P) -> usize
```

Comment:
can you add a comment here stating what the setting of `start_check` entails?

Something like this would suffice (asuming it is an accurate description):
> if `start_check` is false, then the caller has already ensured that  the start of the range does not reproduce the regression (i.e. that running the `predicate` at the start date returns `Satisfies::No`). If `start_check` is true, then this method will itself confirm that running  the `predicate` at the start date returns `Satisfies::No`.

Reply:
Oh,  I now see that this parameter may go away in a later commit.

Assuming the `start_check` parameter does go away in a later commit, then there is no need to document it here.

Reply:
That new parameter that was in early commits in this PR should be gone after the refactor beginning at ~https://github.com/rust-lang/cargo-bisect-rustc/pull/62/commits/e50052ec966ac6d58c28d02dd1572690ef751b11

I was initially using it to prevent the duplicated nightly start date checks but I ended up refactoring the entire approach based on Santiago's suggestion.